frontend/components/forms/checkbox/component.tsx

```diff
@@ -1,34 +1,52 @@
-import { FC } from 'react';
+import React, { useRef, useState } from 'react';
+
+import { useFocusRing } from 'react-aria';
+import { FieldValues } from 'react-hook-form';
 
 import cx from 'classnames';
 
-import useStatus from '../utils';
+import { useCheckbox } from '@react-aria/checkbox';
+import { useToggleState } from '@react-stately/toggle';
+
+import { CheckboxProps } from './types';
 
-import { THEME } from './constants';
-import type { CheckboxProps } from './types';
+export const Checkbox = <FormValues extends FieldValues>(props: CheckboxProps<FormValues>) => {
+  const { register, registerOptions, ...rest } = props;
+  const { name, id, 'aria-label': ariaLabel, className } = rest;
 
-export const Checkbox: FC<CheckboxProps> = ({
-  theme = 'dark',
-  input,
-  meta = {},
-  disabled = false,
-  className,
-  ...props
-}: CheckboxProps) => {
-  const st = useStatus({ meta, disabled });
+  const [invalid, setInvalid] = useState(false);
+  const state = useToggleState(props);
+  const { isSelected, toggle } = state;
+  const { isFocusVisible, focusProps } = useFocusRing(rest);
 
+  const ref = useRef();
+  const { inputProps } = useCheckbox(rest, state, ref);
+  console.log(isFocusVisible);
   return (
     <input
```

Comment:
Please specify the type of the input: `type="checkbox"`. It's very important for accessibility.